html/browsers/browsing-the-web/overlapping-navigations-and-traversals/resources/helpers.mjs

```diff
@@ -39,7 +39,13 @@ export function waitForPopstate(obj) {
 // of a separate iframe load, we would have caught any problems.
 export async function waitForPotentialNetworkLoads(t) {
   const before = performance.now();
+
+  // Sometimes we're doing something, like a traversal, which cancels our first
+  // attempt at iframe loading. In that case we bail out after 100 ms and try
+  // again. (Better ideas welcome...)
+  await Promise.race([createIframe(t), delay(t, 100)]);
   await createIframe(t);
+
   const after = performance.now();
   await delay(t, after - before);
```

Comment:
The original documentation says this helper waits 2x iframe loads, but now I think it waits up to 4x:
 - (a) Race: createIframe() vs 100ms
 - (b) Another createIframe()
 - (c) The time it took to do (a) + (b)

4x in the case where createIframe() is < 100ms; otherwise, 2x + 200ms I guess. Heh that's fine, maybe let's just update the documentation to say that this method waits some "small multiple of an iframe network load"?

Reply:
Yeah... I wish I had a better solution for this.